server/src/main/java/org/elasticsearch/search/rescore/RescorePhase.java

```diff
@@ -21,8 +22,8 @@
  */
 public class RescorePhase {
 
-    public static void execute(SearchContext context) {
-        TopDocs topDocs = context.queryResult().topDocs().topDocs;
+    public static void execute(SearchContext context, SingleSearchResult singleSearchResult) {
+        TopDocs topDocs = singleSearchResult.topDocs().topDocs;
```

Comment:
This should be modified IMO to accept a context and return top docs (or null) and the caller can figure out what to do with it. This may make the reranker execution stuff easier.

Reply:
This is changed as described above where it's basically just an additional list of top docs.